src/components/__tests__/api-provider.test.tsx

```diff
@@ -38,14 +40,19 @@ jest.mock('../../libraries/google-maps-api-loader', () => {
       onLoadingStatusChange(APILoadingStatus.LOADING);
 
       google.maps.importLibrary = importLibraryMock;
-      return new Promise(
-        resolve =>
-          (triggerMapsApiLoaded = () => {
+
+      return new Promise((resolve, reject) => {
+        if (shouldRejectLoad) {
+          reject(new Error('Test error: Failed to load Google Maps API'));
+        } else {
```

Comment:
I would implement this the same way as the success-case is done, so, for example:

```tsx
triggerLoadingFailed = () => { 
  reject();
  onLoadingStatusChange(APILoadingStatus.FAILED);
};
```

then we get the same async behaviour as in the success tests.